Source/WebCore/css/CSSSelector.cpp

```diff
@@ -1042,6 +1047,13 @@ bool CSSSelector::hasExplicitNestingParent() const
     auto checkForExplicitParent = [] (const CSSSelector& selector) {
         if (selector.match() == CSSSelector::Match::NestingParent)
             return true;
+        else if (selector.match() == CSSSelector::Match::UnknownToSerialize) {
+            CSSTokenizer tokenizer(selector.value());
+            for (auto& token : tokenizer.tokenRange()) {
+                if (token.type() == DelimiterToken && token.delimiter() == '&')
+                    return true;
+            }
+        }
         return false;
```

Comment:
We don't want CSSSelector invoking CSSTokenizer. That's inversion of roles.

Reply:
And I suspect that can be fixed by doing this during the parsing and recording it as `UnknownToSerializeNestContaining` as we discussed above.

Reply:
Fixed, we store the "nest containing" information during parsing via the new enum match member.